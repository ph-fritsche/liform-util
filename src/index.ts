export { isShallowEqual } from './comparison/isShallowEqual'
export { intervalFromString, Interval } from './date/interval'
export { getIndicesOfDescendant } from './dom/indicesOfDescendant'
export { useBoundFunction } from './react/boundFunction'
export { useForkedCallback } from './react/forkedCallback'
export { useForkedRef, updateRef } from './react/forkedRef'
export { useId } from './react/id'
export { ComponentForwardRef, WrapperProps, ForwardRefWrapper, createForwardRefWrapper } from './react/wrapper'
export { isEscaped, countUnescaped, hasUnescaped, indexOfUnescaped } from './string/escaped'
export { filterObject } from './object/filterObject'
