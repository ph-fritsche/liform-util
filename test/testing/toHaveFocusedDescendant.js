import '../../src/testing'

it('Pass for active element and its parents', () => {
    document.body.innerHTML = `
        <div tabIndex='0' id='el0'>
            <div tabIndex='0' id='el1'/>
        </div>
        <div tabIndex='0' id='el2'/>
    `

    document.querySelector('#el1').focus()

    expect(() => expect(document.querySelector('#el1')).toHaveFocusedDescendant()).not.toThrowError()
    expect(() => expect(document.querySelector('#el0')).toHaveFocusedDescendant()).not.toThrowError()

    expect(() => expect(document.querySelector('#el2')).toHaveFocusedDescendant()).toThrowErrorMatchingSnapshot()
})
