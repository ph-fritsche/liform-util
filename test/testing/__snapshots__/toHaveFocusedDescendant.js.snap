// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pass for active element and its parents 1`] = `
"expect(DIV).toHaveFocusedDescendant(undefined)

Expected,  <div id=\\"el2\\" tabindex=\\"0\\">
    </div>,Received,  <div id=\\"el1\\" tabindex=\\"0\\">
        </div>"
`;
